localization/pose_estimator_arbiter/src/pose_estimator_arbiter/switch_rule/map_based_rule_core.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "pose_estimator_arbiter/rule_helper/grid_info.hpp"
+#include "pose_estimator_arbiter/switch_rule/map_based_rule.hpp"
+
+namespace pose_estimator_arbiter::switch_rule
+{
+std::unordered_map<PoseEstimatorType, bool> MapBasedRule::update()
+{
+  auto ret = update_impl();
+  RCLCPP_DEBUG(get_logger(), debug_string_.c_str());
```

Comment:
I can avoid the compile error 
```
/opt/ros/humble/include/rcutils/rcutils/logging_macros.h:72:18: error: format not a string literal and no format arguments [-Werror=format-security]
```
by changing this row as the following. (adding "%s",)
```
RCLCPP_DEBUG(get_logger(), "%s", debug_string_.c_str());
```

Is there any problem for doing this?

Reply:
In my environment, it does not cause the compilation error. :thinking:  But I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/5846/commits/80b6c20eef552e1a3b5176c1bebc5c43da95b827